composio/sdk/sdk.py

```diff
@@ -312,84 +313,60 @@ def get_app(self, app_name: str):
-                    return actions_response["items"]
-                for item in actions_response["items"]:
-                    if item["tags"] and any(tag in item["tags"] for tag in tag_values):
-                        filtered_actions.append(item)
-                return filtered_actions
+        app_unique_ids = [app.value for app in apps]
+        action_unique_ids = [action.value[1] for action in actions]
 
+        if app_unique_ids and (not tags or Tag.ALL in tags):
             warnings.warn(
                 "Using all the actions of an app is not recommended. "
                 "Please use tags to filter actions or provide specific actions. "
                 "We just pass the important actions to the agent, but this is not meant "
                 "to be used in production. Check out https://docs.composio.dev/sdk/python/actions for more information.",
                 UserWarning,
             )
-            actions = actions_response["items"]
-            important_tag = Tag.IMPORTANT.value[1]
-            important_actions = [
-                action
-                for action in actions
-                if important_tag in (action.get("tags", []) or [])
-            ]
-            if len(important_actions) > 5:
-                return important_actions
-            else:
-                return actions
+
+        all_relevent_app_ids = app_unique_ids + [action.value[0] for action in actions]
+        all_relevent_app_ids = list(set(all_relevent_app_ids))
+
+        if tags:
+            if actions:
+                raise ValueError("Both actions and tags cannot be provided together")
+            tag_values = [tag.value[1] if isinstance(tag, Tag) else tag for tag in tags]
+        # else:
+        #     tag_values = [Tag.IMPORTANT.value[1]]
+
+        action_response = self.http_handler.get_action_schemas(
+            app_unique_ids=all_relevent_app_ids, use_case=use_case, limit=limit
+        )
+        action_schema_list = action_response["items"]
+        filtered_actions = []
+        important_actions = []
+        important_tag = Tag.IMPORTANT.value[1]
+
+        for action_schema in action_schema_list:
+            if action_schema["appName"] in app_unique_ids:
+                if not tags:
+                    filtered_actions.append(action_schema)
+                elif bool(set(tag_values) & set(action_schema["tags"])):
+                    filtered_actions.append(action_schema)
+
+                if important_tag in action_schema["tags"]:
+                    important_actions.append(action_schema)
+
+            elif action_schema["name"] in action_unique_ids:
+                filtered_actions.append(action_schema)
+                important_actions.append(action_schema)
+
+        if len(important_actions) > 5 and Tag.ALL not in tags:
+            return important_actions
         else:
-            raise ValueError("Either apps or actions must be provided")
+            return filtered_actions
```

Comment:
What if a proper filter is given like Tag.Repo in that case also above condition will return important _actions. Fix it.

Reply:
Solved.